src/values.rs

```diff
@@ -109,9 +133,50 @@ impl Ord for OnOff {
     }
 }
 
-/// An open/closed state. Internal representation may be either
-/// Open or Closed.
-#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+impl Serialize for OnOff {
+    fn serialize<S>(&self, serializer: &mut S) -> Result<(), S::Error> where S: Serializer {
+        match *self {
+            OnOff::On => "On".serialize(serializer),
+            OnOff::Off => "Off".serialize(serializer)
+        }
+    }
+}
+impl Deserialize for OnOff {
+    fn deserialize<D>(deserializer: &mut D) -> Result<Self, D::Error> where D: Deserializer {
+        deserializer.deserialize_string(TrivialEnumVisitor::new(|source| {
+            match source {
+                "On" | "on" | "ON" => Ok(OnOff::On),
```

Comment:
__question:__ I have a mixed feelings on this, from one side this approach looks a bit fragile (e.g. for multiple-words variants like ```MotionDetected``` it's not that obvious what formats we should support), from the other one good error message should help developer to figure out if format he's using is wrong. Do you know if something like ```source.to_ascii_upper()``` have too big memory impact for our use cases?